### PR TITLE
Alert the user when the images in a stack don't all have the same size

### PR DESCRIPTION
**Pull request by alesgenova**
#1507

### CODE REVIEW

File: tomviz/LoadDataReaction.cxx [hunk 1]
```diff
@@ -218,7 +223,22 @@ DataSource* LoadDataReaction::loadData(const QStringList& fileNames,
...
+             info.suffix().toLower() == "tif") {
+    if (fileNames.size() > 1) {
+      // Ensure all the images in the stack have the same size.
+      std::vector<ImageFileInfo> summary;
+      if (!loadTiffStack(fileNames, summary)) {
+        badStackAlert(summary);
+        return nullptr;
+      }
+    }
+    loadWithParaview = true;
```
[the remark this answers is not in the document]
Comment (alesgenova) [reply tomviz/LoadDataReaction.cxx]: I did it like this to ensure that in the future things won't break if another "else if" block is added, but can certainly simplify it if you want

File: tomviz/LoadDataReaction.cxx [hunk 2]
```diff
@@ -13,6 +13,8 @@
   limitations under the License.
 
 ******************************************************************************/
+#include <vector>
```
Comment (cryos) [inline tomviz/LoadDataReaction.cxx]: Try to group includes, this is STL and should be with sstream lower down - alphabetically sorted.

File: tomviz/LoadDataReaction.cxx (hunk 1 above)
[the remark this answers is not in the document]
Comment (cryos) [reply tomviz/LoadDataReaction.cxx]: I would keep it simple

File: tomviz/LoadDataReaction.cxx [hunk 3]
```diff
@@ -256,6 +276,65 @@ DataSource* LoadDataReaction::loadData(const QStringList& fileNames,
   return dataSource;
 }
 
+bool LoadDataReaction::loadTiffStack(const QStringList& fileNames,
+                                     std::vector<ImageFileInfo>& summary)
+{
+
+  vtkSmartPointer<vtkTIFFReader> reader = vtkSmartPointer<vtkTIFFReader>::New();
```
Comment (cryos) [inline tomviz/LoadDataReaction.cxx]: Should prefer vtkNew for this, avoids duplication/long line.

File: tomviz/LoadDataReaction.cxx [hunk 4]
```diff
@@ -256,6 +276,61 @@ DataSource* LoadDataReaction::loadData(const QStringList& fileNames,
...
+    summary[i].fileName = file;
+    summary[i].consistent = true;
+
+    if (n == -1 && m == -1) {
+      n = dims[0];
+      m = dims[1];
+    } else {
+      if (n != dims[0] || m != dims[1]) {
+        summary[i].consistent = false;
+        success = false;
```
[the remark this answers is not in the document]
Comment (cryos) [reply tomviz/LoadDataReaction.cxx]: Yes, we should return on failure, if you get to the end you can declare success.

File: tomviz/LoadDataReaction.h [hunk 5]
```diff
@@ -16,13 +16,17 @@
 #ifndef tomvizLoadDataReaction_h
 #define tomvizLoadDataReaction_h
 
+#include <vector>
```
Comment (cryos) [inline tomviz/LoadDataReaction.h]: I think you should use a QList here, and should just declare the type here, include its header in the implementation.

File: tomviz/LoadDataReaction.cxx (hunk 4 above)
[the remark this answers is not in the document]
Comment (alesgenova) [reply tomviz/LoadDataReaction.cxx]: Fixed in https://github.com/OpenChemistry/tomviz/pull/1535/commits/b22ef95164146a9e14615c6ad726d6dd72e99c0c

File: tomviz/LoadDataReaction.h (hunk 5 above)
Comment (alesgenova) [reply tomviz/LoadDataReaction.h]: Fixed in https://github.com/OpenChemistry/tomviz/pull/1535/commits/aac9bca28336d6a2d404826c0e0b69b8294ed33a

File: tomviz/LoadDataReaction.cxx (hunk 3 above)
Comment (alesgenova) [reply tomviz/LoadDataReaction.cxx]: Fixed in https://github.com/OpenChemistry/tomviz/pull/1535/commits/9883950e6f70d5dd20dc79458084289fec5fdd7f

File: tomviz/LoadDataReaction.cxx (hunk 1 above)
[the remark this answers is not in the document]
Comment (cryos) [reply tomviz/LoadDataReaction.cxx]: Let's simplify this as discussed, and then I think this branch will be ready to merge.

[the remark this answers is not in the document]
Comment (alesgenova) [reply tomviz/LoadDataReaction.cxx]: Fixed in https://github.com/OpenChemistry/tomviz/pull/1535/commits/b033106d0e611d388803a0eebfa15d1f18d31cf8

**Pull request merged**